docs/src/starknet/deploy.md

```diff
@@ -82,6 +82,45 @@ transaction: https://sepolia.starkscan.co/tx/[..]
 > It is important to know how types are serialized because all values passed as constructor calldata are
 > interpreted as a field elements (felt252).
 
+### Deploying and Declaring at the Same Time
+
+You can deploy and declare a contract at the same time by passing `--contract-name` flag instead of `--class-hash`.
+Under the hood, if the passed contract was never declared to starknet, it will run the [declare](../starknet/declare.md)
+command first and then execute the contract deployment.
```

Comment:
I think we should present this as deployment by contract name rather than deploying and declaring at the same time. The automatic declare is just a side effect, which allows for a simpler way to deploy without needing the class hash.

Reply:
I don't think we should be hiding the fact that this makes 2 separate transactions. This is meant to be used as an convenience function, not magic to abstract how starknet works from users.